mockbroker_test.go

```diff
@@ -77,6 +84,10 @@ func (b *mockBroker) BrokerID() int32 {
 	return b.brokerID
 }
 
+func (b *mockBroker) History() []RequestResponse {
+	return b.history
```

Comment:
if further request/responses can update the history slice, should this not be returning a copy (and probably taking the lock as well)?